### PR TITLE
Fixed modal fadeout animations

### PR DESCRIPTION
**Pull request by samturrell**
The fade out feature of the original t4t5/sweetalert modal and backdrop seems to have been removed or broken along the way. 

This commit reimplements this functionality to provide a much smoother experience.

### CODE REVIEW

**Comment by limonte** [comment]
Thank you @samturrell for this PR! I'll leave some comments on your code :)

File: src/sweetalert2.js [hunk 1]
```diff
@@ -1070,6 +1104,19 @@
     removeClass($warningIcon, 'pulse-warning');
 
     resetPrevState();
+    
+    if (animationEndEvent && !hasClass(modal, 'no-animation')) {
+      modal.addEventListener(animationEndEvent, function swalCloseEventFinished() {
+        modal.removeEventListener(animationEndEvent, swalCloseEventFinished);
+        if (hasClass(modal, 'hide-swal2')) {
+          _hide(modal);
+          fadeOut(getOverlay(), 10);
```
Comment (limonte) [inline src/sweetalert2.js]: why `10` and not `0`? Let's hide overlay right after modal hide.

File: src/sweetalert2.scss [hunk 2]
```diff
@@ -452,7 +452,7 @@
   }
 
   100% {
-    transform: scale(.5);
+    transform: scale(0);
```
Comment (limonte) [inline src/sweetalert2.scss]: `0.5` is prettier to me personally, let's keep `.5` here  :)

**Comment by samturrell** [comment]
How's that, @limonte? :)

**Comment by limonte** [comment]
@samturrell Terrific! Thank you!

**Pull request merged**